src/index.ts

```diff
@@ -50,9 +58,20 @@ async function run() {
         ).stdout.includes('filter: lfs');
 
         if (shouldBeStoredInLFS) {
-          const isStoredInLFS = Boolean(
-            file.patch?.includes('version https://git-lfs.github.com/spec/v1')
-          );
+          let isStoredInLFS = false;
+          if (file.patch) {
+            isStoredInLFS = Boolean(
+              file.patch.includes('version https://git-lfs.github.com/spec/v1')
+            );
+          } else if (file.filecontents) {
+            // For PDF files, seems GitHub doesn't have the .patch property set
+            // in the API response. Reading file contents instead
+            // https://github.com/ActionsDesk/lfs-warning/issues/135
+            isStoredInLFS = file.filecontents.startsWith(
+              lfsFileContentStartInBase64
+            );
```

Comment:
nit: not 100% sure this causes any change of behavior ? 🤔 

i.e. if we look at `patch` only, then I guess we'd report a "this file should be in LFS" only in the PR that introduces this new file (i.e. `patch` will contain the newly added first line with `version https:/....` ... in the future, updates to that same file will still have this `version https/...` prefix, so the `version https/....` will maybe not appear in `patch` as it's an "unchanged" part of the existing file)

By looking at `filecontents`, we always look at the full content of the file "as per the branch" of this PR ... which means this would report both new files, and changes to already existing files ... 

It's probably not a big issue though, and maybe my assumptions are wrong 🙃

Reply:
yeah, it's a bit weird because patch in updates to a file shouldn't contain that version line if hasn't changed. however, from experience, seems it works: https://github.com/davidlj95/lfs-warning-test/pull/4

no idea why tho